copy_test.go

```diff
@@ -1790,7 +1792,8 @@ func TestCopyGraph_WithOptions(t *testing.T) {
 			t.Fatalf("CopyGraph() error = %v, wantErr %v", err, e)
 		}
 
-		if got, expected := dst.numExists.Load(), int64(7); got != expected {
+		// with a very low probability, dst.numExists may be 3
```

Comment:
Why?🤔

Reply:
Due to the structure of the graph being copied, there are two possible orders of visiting each node. With a very low probability, one of the orders will make `dst.numExists` 3. Under most of the time the value will be 4.

Reply:
> Due to the structure of the graph being copied, there are two possible orders of visiting each node. With a very low probability, one of the orders will make dst.numExists 3.

Would this impact other unit tests or user code as well?

Reply:
> > Due to the structure of the graph being copied, there are two possible orders of visiting each node. With a very low probability, one of the orders will make dst.numExists 3.
> 
> Would this impact other unit tests or user code as well?

Under our current knowledge, no, this is the only place impacted.

Reply:
Oops, looks like the value can also be 5, under `go 1.21`.
![image](https://github.com/oras-project/oras-go/assets/103478229/6a3894d8-3a74-4ddc-9c35-2543fedc06e5)